File: Week 2/readability/readability.c
#include <stdio.h>
#include "cs50.c"
#include <string.h>
#include <math.h>
#include <ctype.h>

// Alphabets
char *alphabets = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

int main(void)
{
    // Initialize variables
    int words = 1;
    int sentences = 0;
    int letters = 0;

    // Get string
    string text = get_string("Text: ");

    // length of the string
    int n = strlen(text);

    // Count the number of letters
    for (int i = 0; i < n; i++)
    {
        if (strchr(alphabets, toupper(text[i])) != NULL)
        {
            letters++;
        }
    }

    // Count the number of words
    for (int i = 0; i < n; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }

    // Count the number of sentences
    for (int i = 0; i < n; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }

    // Coleman Liau formula
    float result = 0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8;

    // Print result
    if (result <= 0)
    {
        printf("Before Grade 1\n");
    }
    else if (result >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", (int)round(result));
    }
}

File: Week 6/Readability/readability.py
# Intialize
words = 1
sentences = 0
letters = 0

# Alphabets
alphabets = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
# User input
text = input("Text: ")

# Text Length
n = len(text)

# Number of letters
for i in range(n):
    if (text[i].upper() in alphabets):
        letters += 1
    else:
        pass

# Number of words 
for i in range(n):
    if (text[i] == " "):
        words += 1
    else:
        pass

# Number of sentences 
for i in range(n):
    if (text[i] == '.' or text[i] == '?' or text[i] == '!'):
        sentences += 1
    else:
        pass

# Result
result = 0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8

# Print Result
if (result <= 0):

    print("Before Grade 1\n")

elif (result >= 16):

    print("Grade 16+\n")

else:
    print(f"Grade {round(result)}\n")
